zhusuan/distributions/multivariate.py

```diff
@@ -60,38 +66,11 @@ def __init__(self,
             dtype = tf.int32
         assert_same_float_and_int_dtype([], dtype)
 
-        static_logits_shape = self._logits.get_shape()
-        shape_err_msg = "logits should have rank >= 1."
-        if static_logits_shape and (static_logits_shape.ndims < 1):
-            raise ValueError(shape_err_msg)
-        elif static_logits_shape and (
-                static_logits_shape[-1].value is not None):
-            self._n_categories = static_logits_shape[-1].value
-        else:
-            _assert_shape_op = tf.assert_rank_at_least(
-                self._logits, 1, message=shape_err_msg)
-            with tf.control_dependencies([_assert_shape_op]):
-                self._logits = tf.identity(self._logits)
-            self._n_categories = tf.shape(self._logits)[-1]
-
-        sign_err_msg = "n_experiments must be positive"
-        if isinstance(n_experiments, int):
-            if n_experiments <= 0:
-                raise ValueError(sign_err_msg)
-            self._n_experiments = n_experiments
-        else:
-            try:
-                n_experiments = tf.convert_to_tensor(n_experiments, tf.int32)
-            except ValueError:
-                raise TypeError('n_experiments must be int32')
-            _assert_rank_op = tf.assert_rank(
-                n_experiments, 0,
-                message="n_experiments should be a scalar (0-D Tensor).")
-            _assert_positive_op = tf.assert_greater(
-                n_experiments, 0, message=sign_err_msg)
-            with tf.control_dependencies([_assert_rank_op,
-                                          _assert_positive_op]):
-                self._n_experiments = tf.identity(n_experiments)
+        self._logits, self._n_categories = assert_rank_at_least_one(
+            self._logits, 'Multinomial.logits')
+
+        self._n_experiments = assert_positive_integer(
+            n_experiments, tf.int32, 'Multinomial.n_experiments')
```

Comment:
sorry but maybe assert_positive_int32_scalar is better without the int32 parameter